src/poetry/repositories/http_repository.py

```diff
@@ -211,11 +245,39 @@ def _links_to_data(self, links: list[Link], data: PackageInfo) -> dict[str, Any]
                 f' "{data.version}"'
             )
         urls = defaultdict(list)
+        metadata = {}
```

Comment:
Could you add explicit typing? It makes the code easier to follow
```suggestion
        metadata: dict[str, pkginfo.Distribution] = {}
```